### PR TITLE
[WiP] Additionnal features and methods for Mesh module : Issue #48

### PR DESCRIPTION
**Pull request by Superomeg4**
Some bug fixes
New features to add elements to an existing ElementDict object
New features to compute the interface between two meshes (Example with a mesh of Segments at the interface of two meshes of Triangles)

### CODE REVIEW

File: Methods/Mesh/Mesh/interface.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+from pyleecan.Classes.ElementDict import ElementDict
+from pyleecan.Classes.NodeMat import NodeMat
+from collections import Counter
+import numpy as np
+
+
+def interface(self, other_mesh):
```
Comment (Superomeg4) [inline Methods/Mesh/Mesh/interface.py]: Maye we should split this method in several function.

**Comment by Superomeg4** [comment]
Adding the node_tag attribute for NodeMat classes. Otherwise it is going to be to complicated to manage submeshes.

**Comment by Superomeg4** [comment]
Modifications from discussion #48 have been taken into account.

File: Classes/Mesh.py [hunk 2]
```diff
@@ -61,14 +182,43 @@ def __init__(self, element=None, node=None, submesh=list(), init_dict=None):
                 submesh = init_dict["submesh"]
         # Initialisation by argument
         self.parent = None
-        # element can be None, a Element object or a dict
-        if isinstance(element, dict):
-            self.element = Element(init_dict=element)
+        # element can be None or a list of Element object
```
Comment (BonneelP) [inline Classes/Mesh.py]: a list => a dict

Comment (BonneelP) [inline Classes/Mesh.py]: a list => a dict

File: Classes/Mesh.py [hunk 3]
```diff
@@ -61,14 +182,43 @@ def __init__(self, element=None, node=None, submesh=list(), init_dict=None):
...
+                            + " in init_dict for element"
+                        )
+                    # Dynamic import to call the correct constructor
+                    module = __import__(
+                        "pyleecan.Classes." + class_name, fromlist=[class_name]
+                    )
+                    class_obj = getattr(module, class_name)
+                    self.element[key] = class_obj(init_dict=obj)
+                else:
+                    element = element  # Should raise an error
```
Comment (BonneelP) [inline Classes/Mesh.py]: else:
    self.element[key] = obj

File: Classes/Mesh.py [hunk 4]
```diff
@@ -97,10 +247,18 @@ def __str__(self):
             Mesh_str += "parent = None " + linesep
         else:
             Mesh_str += "parent = " + str(type(self.parent)) + " object" + linesep
-        if self.element is not None:
-            Mesh_str += "element = " + str(self.element.as_dict()) + linesep + linesep
-        else:
-            Mesh_str += "element = None" + linesep + linesep
+        if len(self.element) == 0:
+            Mesh_str += "element = []"
```
Comment (BonneelP) [inline Classes/Mesh.py]:
```suggestion
            Mesh_str += "element = {}"
```

**Comment by BonneelP** [comment]
Thank you for your contribution :)

**Pull request merged**